### PR TITLE
Add e2e test script for lambda deployment

### PR DESCRIPTION
**Pull request by yubozhao**
`python ./script/e2e_lambda_deployment.py`

### CODE REVIEW

File: scripts/e2e_lambda_deployment.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                headers={'Content-Type': 'application/json'},
+            )
+            if request_result.status_code != 200:
+                deployment_failed = True
+            if request_result.content.decode('utf-8') != '[0]':
+                deployment_failed = True
+        except Exception as e:
+            logger.error(str(e))
+            deployment_failed = True
+        test_endpoint_command = [
```
Comment (parano) [inline scripts/e2e_lambda_deployment.py]: not used?

File: scripts/e2e_lambda_deployment.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+@artifacts([PickleArtifact('clf')])
+@env(pip_dependencies=['scikit-learn'])
+class IrisClassifier(BentoService):
+    @api(DataframeHandler)
+    def predict(self, df):
+        return self.artifacts.clf.predict(df)
+
+
+if __name__ == '__main__':
+    print('E2E DEPLOYMENT TEST FOR LAMBDA:::: Training iris classifier')
```
Comment (parano) [inline scripts/e2e_lambda_deployment.py]: use logger.info?, change logger to logger = logging.getLogger("bentoml.test")
 ?

File: scripts/e2e_lambda_deployment.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    loaded_service = load(saved_path)
+    sample_data = X[0:1]
+
+    deployment_failed = False
+    print(
+        'E2E DEPLOYMENT TEST FOR LAMBDA:::: Creating AWS Lambda test deployment '
+        'for iris classifier with BentoML CLI'
+    )
+    bento_name = '{}:{}'.format(loaded_service.name, loaded_service.version)
+    deployment_name = 'tests-lambda-e2e'
```
Comment (parano) [inline scripts/e2e_lambda_deployment.py]: let's add a random string to this name - note that we may move this to CI later

File: tests/deployment/aws_lambda/test_aws_lambda_deployment_operator.py [hunk 4]
```diff
@@ -104,6 +104,32 @@ def return_predict_func(mock_load_service, mock_download_artifacts):
...
+    open(os.path.join(mock_bento_bundle_path, 'requirements.txt'), 'w').close()
+
+    init_sam_project(
+        mock_sam_project_path,
+        mock_bento_bundle_path,
+        mock_deployment_name,
+        mock_bento_name,
+        mock_api_names,
+    )
+    assert os.path.isfile(
```
Comment (parano) [inline tests/deployment/aws_lambda/test_aws_lambda_deployment_operator.py]: could you also add assertions for other init_sam_projec behaviors?

File: scripts/e2e_lambda_deployment.py (hunk 1 above)
Comment (yubozhao) [reply scripts/e2e_lambda_deployment.py]: yeah, we just use `requests` package instead. removing it now

**Pull request merged**